api/filter_dimension_options.go

```diff
@@ -517,23 +517,46 @@ func (api *FilterAPI) patchFilterBlueprintDimensionHandler(w http.ResponseWriter
 	}
 
 	// apply the patches to the filter blueprint dimension options
-	successfulPatches, newETag, err := api.patchFilterBlueprintDimension(ctx, filterBlueprintID, dimensionName, patches, logData, eTag)
+	var newETag interface{}
+	newETag, err = api.dataStore.RunTransaction(ctx, true, func(transactionCtx context.Context) (interface{}, error) {
+		var (
+			allOptions, options []string
+			tag                 = eTag
+		)
+
+		// apply patch operations sequentially, stop processing if one patch fails, and return a list of successful patches operations
+		for _, patch := range patches {
+			allOptions, err = getStringArrayFromInterface(patch.Value)
+			if err != nil {
+				return tag, err
+			}
+			options = RemoveDuplicateAndEmptyOptions(allOptions)
+
+			if patch.Op == dprequest.OpAdd.String() {
+				tag, err = api.addFilterBlueprintDimensionOptions(transactionCtx, filterBlueprintID, dimensionName, options, logData, tag)
+				if err != nil {
+					return tag, err
+				}
+			} else {
+				tag, err = api.removeFilterBlueprintDimensionOptions(transactionCtx, filterBlueprintID, dimensionName, options, logData, tag)
+				if err != nil {
+					return tag, err
+				}
+			}
+		}
+
+		return tag, nil
+	})
 	if err != nil {
-		logData["successful_patches"] = successfulPatches
 		log.Error(ctx, "error patching filter blueprint dimension options", err, logData)
 		setErrorCodeFromError(w, err)
-		if len(successfulPatches) > 0 {
-			if err := WriteJSONBody(ctx, successfulPatches, w, logData); err != nil {
-				log.Error(ctx, "error writing JSON body during filter blueprint patch error handling", err, logData)
-			}
-		}
 		return
 	}
 
 	// set content type, marshal and write response
 	setJSONPatchContentType(w)
-	setETag(w, newETag)
-	if err := WriteJSONBody(ctx, successfulPatches, w, logData); err != nil {
+	setETag(w, newETag.(string))
```

Comment:
Probably best not to presume the interface will cast correctly here as it will panic if it doesn't. To check the underlying value is a string you can do something like:

```
	if s, ok := newEtag.(string); ok {
		setEtag(s)
	} else {
		// handle error/default etc
	}
```

Reply:
yeah was relying on the fact that it is going to be a valid string if no error from RunTransaction, but I will add the guard